src/test/java/support/fakes/FakeStorageModule.java

```diff
@@ -294,22 +294,34 @@ private void checkTokenHeader(RoutingContext routingContext) {
   private void checkRequiredProperties(RoutingContext routingContext) {
     JsonObject body = getJsonFromBody(routingContext);
 
-    ArrayList<ValidationError> errors = new ArrayList<>();
+    List<ValidationError> errors = new ArrayList<>();
 
-    requiredProperties.stream().forEach(requiredProperty -> {
-      if(!body.getMap().containsKey(requiredProperty)) {
+    requiredProperties.forEach(requiredProperty -> {
+      if (getPropertyValue(body, requiredProperty) == null) {
         errors.add(new ValidationError("Required property missing", requiredProperty, ""));
       }
     });
 
-    if(errors.isEmpty()) {
+    if (errors.isEmpty()) {
       routingContext.next();
-    }
-    else {
+    } else {
       JsonResponse.unprocessableEntity(routingContext.response(), errors);
     }
   }
 
+  private Object getPropertyValue(JsonObject body, String requiredProperty) {
+    String[] pathElements = requiredProperty.split("\\.");
```

Comment:
I think we should consider using something like JSON path for checks like this rather than our own format. This is likely beyond the scope of this work

Reply:
Agreed.